lib/utils/visFile.cpp

```diff
@@ -106,7 +107,12 @@ void visFileBundle::add_file(time_ctype first_time) {
 
     // Construct the name of the new file
     uint32_t time_since_start = (uint32_t)(first_time.ctime - acq_start_time);
-    std::string file_name = fmt::format(fmt("{:08d}_{:04d}"), time_since_start, freq_chunk);
+    std::string file_name;
+    if (acq_type == "hfb") {
```

Comment:
I'm not sure about this part. It's taken some code that has been carefully kept generic and added an if statement that dispatches on acq type. It feels a bit out of place.

I've been trying to think of ways around this, and I can see two. The first is to let the file type implementation decide what it wants to be called, and have it change the name in there. That mostly works, but requires some breaking apart of the path that the file constructors get.

What I think is a much better solution is to move the policy of the file naming up into the writer task. This both simplifies some existing things, and makes this possible. I think you do something like:
- Assign `acqname_fmt` and `filename_fmt` params into visFileBundle. These replace the hardcoded fmt strings used to currently create those names and should accept a short set of named params to create the format (those named params are provided using `fmt::arg`).
- This also allows removing the instrument name and freq chunk arguments from visFileBundle as they are only used to create the acq name.
- Define the acq and file name templates in the Writer classes. The vis and hfb writers should use the same template for now (but in both cases they will directly insert the instrument name into it), but they can diverge on the file name template.

Reply:
So have something like:
```
    // Start the acq and create the directory if required
    if (acq_name.empty()) {
        // Format the time (annoyingly you still have to use streams for this)
        acq_name =
            fmt::format("{:%Y%m%dT%H%M%SZ}_{:s}", *std::gmtime(&t), acqname_fmt);
        // Set the acq fields on the instance                                                                                                                                                                      
        acq_start_time = first_time.ctime;

        // Create acquisition directory. Don't bother checking if it already exists, just let it
        // transparently fail
        mkdir((root_path + "/" + acq_name).c_str(), 0775);
    }
```
for the `acq_name`?

What about the filename though:
```
   uint32_t time_since_start = (uint32_t)(first_time.ctime - acq_start_time);
    std::string file_name;
    if (acq_type == "hfb") {
        file_name = fmt::format(fmt("hfb_{:08d}_{:04d}"), time_since_start, freq_chunk);
    } else {
        file_name = fmt::format(fmt("{:08d}_{:04d}"), time_since_start, freq_chunk);
    }
```
Would you add the `time_since_start` to HFB and Vis Writer also?